flocker/node/agents/gce.py

```diff
@@ -544,3 +549,42 @@ def destroy_volume(self, blockdevice_id):
             else:
                 raise e
         return None
+
+    def list_live_nodes(self):
+        page_token = None
+        done = False
+        nodes = []
+        while not done:
+            result = self._compute.instances().list(
+                project=self._project,
+                zone=self._zone,
+                maxResults=self._page_size,
+                pageToken=page_token
+            ).execute()
+            page_token = result.get('nextPageToken')
+            nodes.extend(result.get('items', []))
+            done = not page_token
+        return set(node.get("name") for node in nodes
+                   if node.get("status") == "RUNNING")
```

Comment:
:question: Might be better to use `node["name"]` and `node["status"]` so we get an clear error if the keys of the node dictionary ever change.